### PR TITLE
Add `include_asm` and `include_hex`

### PR DESCRIPTION
**Pull request by lightclient**
I've really go back and forth a lot on how to do this. At this point, I sort of just wanted to get it done. Not very happy with the code, but it does work. Let me know what you think. I'm sort of thinking I'll try to also hack out macros soon and then freeze features for a bit and really clean up the assembler.

Btw: better name recommendations that "node"?

### CODE REVIEW

**Comment by lightclient** [comment]
closes #4

File: etk-asm/src/asm.rs [hunk 1]
```diff
@@ -15,23 +35,33 @@ impl From<TryFromIntError> for Error {
     }
 }
 
-#[derive(Debug, Default)]
```
Comment (SamWilsn) [inline etk-asm/src/asm.rs:18]: Why remove default?

File: etk-asm/src/asm.rs [hunk 2]
```diff
@@ -59,68 +89,121 @@ impl Assembler {
...
+                            let realized = op.realize(s)?;
+                            self.pending.pop_front();
+                            Node::Op(realized)
+                        }
+                        None => self.pending.pop_front().unwrap(),
+                    };
+
+                    popped.assemble(&mut self.ready);
                 }
+                _ => unimplemented!(),
```
Comment (SamWilsn) [inline etk-asm/src/asm.rs:205]: Why does `push_pending` take a `Node` if anything other than an `Op` is unimplemented?

File: etk-asm/src/asm.rs [hunk 3]
```diff
@@ -179,4 +282,59 @@ mod tests {
...
+        );
+        let nodes = nodes![
+            Op::Push1(Imm::from(1)),
+            Node::IncludeAsm(f.path().to_owned()),
+            Op::Push1(Imm::from(2)),
+        ];
+        let mut asm = Assembler::new();
+        let sz = asm.push_all(nodes)?;
+        assert_eq!(sz, 9);
+        assert_eq!(asm.take(), hex!("60015b586000566002"));
```
Comment (SamWilsn) [inline etk-asm/src/asm.rs:320]: Shouldn't this be `60015b586002566002`?

**Review by SamWilsn — CHANGES_REQUESTED** [review]
Added two test cases that I don't think are handled properly. What do you think?

File: etk-asm/src/asm.rs (hunk 1 above)
Comment (lightclient) [reply etk-asm/src/asm.rs:18]: Adding back, not sure why I was thinking I needed to remove it.

File: etk-asm/src/asm.rs (hunk 2 above)
Comment (lightclient) [reply etk-asm/src/asm.rs:205]: I should rewrite that portion to be `if let Node::Op(op) = next`... but `self.pending.push_back(node)` occurs for things other than op so it should include `node`.

File: etk-asm/src/asm.rs (hunk 3 above)
Comment (lightclient) [reply etk-asm/src/asm.rs:320]: I think the label should resolve to `0` because `include_asm` is including *assembled* ops, e.g. for initcode.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**